### PR TITLE
Hide docker file section when runtime != custom in flex facet project structure form

### PR DESCRIPTION
**Pull request by nbashirbello**
Fixes part of #1334 that references hiding the docker file section when runtime != custom.

Not does address location to the docker file section as discussed in #1334.

### CODE REVIEW

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/FlexibleFacetEditor.java [hunk 1]
```diff
@@ -234,15 +234,20 @@ private void showWarnings() {
 
   private void toggleDockerfileSection() {
     boolean enabled = false;
+    boolean visible = true;
```
Comment (etanshaul) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/FlexibleFacetEditor.java]: do we even need this variable anymore? it looks like it just gets assigned to enabled

Comment (nbashirbello) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/FlexibleFacetEditor.java]: We need it when a MalformedYamlFileException is thrown.

Comment (etanshaul) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/FlexibleFacetEditor.java]: not sure I follow; looks like when the exception is thrown its just ignored. I think its valid to say that if the parsing fails, then we should hide it entirely (not have it visible, but disabled)

**Comment by patflynn** [comment]
@nbashirbello thanks nabila!

When you say 'fixes part of..' could you see which part? :)

Also for changes that affect the UI before and after screenshots are super helpful.

**Comment by nbashirbello** [comment]
@patflynn 

I have updated the comments of the PR to reflect your suggestions. I will include screenshots when I  adjust the UI.

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/FlexibleFacetEditor.java [hunk 2]
```diff
@@ -234,16 +234,15 @@ private void showWarnings() {
...
-      enabled = isRuntimeCustom();
-      // Shows no Dockerfile label if runtime is Java and appYaml is valid.
-      noDockerfileLabel.setVisible(!enabled && isValidConfigurationFile(appYaml.getText()));
+      visible = isRuntimeCustom();
     } catch (MalformedYamlFileException myf) {
-      noDockerfileLabel.setVisible(false);
+      // do nothing
     }
-    dockerfile.setEnabled(enabled);
-    genDockerfileButton.setEnabled(enabled);
```
Comment (etanshaul) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/FlexibleFacetEditor.java]: are these enabled by default always? or should we explicitly setEnabled still?

Comment (nbashirbello) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/FlexibleFacetEditor.java]: Yes, when you add a element to a form, it is enabled by default

**Pull request merged**